tasks/Task.cpp

```diff
@@ -13,37 +87,528 @@ Task::~Task()
 {
 }
 
+bool Task::configureHook()
+{
+    if (!TaskBase::configureHook())
+        return false;
+
+    std::string file_name = manageJsonFile();
+    if (!LivoxLidarSdkInit(file_name.c_str())) {
+        LOG_ERROR_S << "Failed to connect to Lidar!" << std::endl;
+        LivoxLidarSdkUninit();
+        return false;
+    }
 
+    m_command_completed = false;
+    SetLivoxLidarInfoChangeCallback(lidarInfoChangeCallback, this);
+    waitForCommandSuccess();
 
-/// The following lines are template definitions for the various state machine
-// hooks defined by Orocos::RTT. See Task.hpp for more detailed
-// documentation about them.
+    applyCommand(QueryLivoxLidarInternalInfo(handle, queryInternalInfoCallback, this));
 
-bool Task::configureHook()
-{
-    if (! TaskBase::configureHook())
+    if (!configureLidar()) {
+        LOG_ERROR_S << "Failed to configure the lidar!" << std::endl;
         return false;
+    }
+    LOG_INFO_S << "Finished configuration process!!!" << endl;
     return true;
 }
+
 bool Task::startHook()
 {
-    if (! TaskBase::startHook())
+    if (!TaskBase::startHook())
         return false;
+
+    updatePointcloudReceivedStatus(false);
```

Comment:
You don't need that here. Point clouds are supposed to be received periodically (unlike commands that are one-shot), so it would work (and be clearer multi-threading wise) to set the flag to false at the beginning of waitForPointcloudSuccess